network/src/commonMain/kotlin/com/wire/kalium/network/exceptions/KaliumException.kt

```diff
@@ -1,27 +1,8 @@
 package com.wire.kalium.network.exceptions
 
-import kotlinx.serialization.Serializable
-
-// TODO: return custom error instead of throwing exception
-
-@Serializable
-sealed class KaliumException : Exception()
-
-@Serializable
-sealed class NetworkException(
-    open var _message: String? = null,
-    open var _code: Int = 0,
-    open var _label: String? = null
-) : KaliumException()
-
-@Serializable
-class AuthException : NetworkException {
-    constructor(message: String?, code: Int) : super(_message = message, _code = code)
-    constructor(code: Int) : super(_code = code)
-
-    constructor(
-        message: String?,
-        code: Int,
-        label: String?
-    ) : super(message, code, label)
+sealed class KaliumException : Exception() {
+    class RedirectError(code: Int, throwable: Throwable?): KaliumException()
+    class InvalidRequestError(code: Int, throwable: Throwable?): KaliumException()
+    class ServerError(code: Int, throwable: Throwable?): KaliumException()
```

Comment:
server errors return something like this in the body for error code 401 - 499:
{
  "code": 403,
  "message": "The user has no verified identity (email or phone number).",
  "label": "no-identity"
}
so preserving this info is usefull
there is an ErrorResponse data class in legacy that you can use